### PR TITLE
Log authentication exceptions in debug mode

### PR DESCRIPTION
**Pull request by arjantijms**
Signed-off-by: Arjan Tijms <arjan.tijms@gmail.com>

### CODE REVIEW

File: appserver/security/webintegration/src/main/java/com/sun/web/security/RealmAdapter.java [hunk 1]
```diff
@@ -1346,14 +1348,10 @@ private boolean validate(HttpRequest request, HttpResponse response, LoginConfig
...
-            if (_logger.isLoggable(FINE)) {
-                _logger.log(FINE, "Jakarta Authentication: http msg authentication fail", ae);
-            }
+            _logger.log(debug? SEVERE : FINE, "Jakarta Authentication: http msg authentication fail", ae);
             httpServletResponse.setStatus(SC_INTERNAL_SERVER_ERROR);
         } catch (RuntimeException e) {
-            if (_logger.isLoggable(FINE)) {
-                _logger.log(FINE, "Jakarta Authentication: Exception during validateRequest", e);
-            }
+            _logger.log(debug? SEVERE : FINE, "Jakarta Authentication: Exception during validateRequest", e);
```
Comment (dmatej) [inline appserver/security/webintegration/src/main/java/com/sun/web/security/RealmAdapter.java:1354]: I would log runtime exceptions as severe and not depending on enabled debugger

File: appserver/security/webintegration/src/main/java/com/sun/web/security/RealmAdapter.java [hunk 2]
```diff
@@ -1346,14 +1348,10 @@ private boolean validate(HttpRequest request, HttpResponse response, LoginConfig
                 httpServletRequest.setAttribute(MESSAGE_INFO, messageInfo);
             }
         } catch (AuthException ae) {
-            if (_logger.isLoggable(FINE)) {
-                _logger.log(FINE, "Jakarta Authentication: http msg authentication fail", ae);
-            }
+            _logger.log(debug? SEVERE : FINE, "Jakarta Authentication: http msg authentication fail", ae);
```
Comment (dmatej) [inline appserver/security/webintegration/src/main/java/com/sun/web/security/RealmAdapter.java:1351]: I would log authexceptions as FINE and not depending on debugger.

**Comment by arjantijms** [comment]
2nd round ;)

**Review by dmatej — APPROVED** [review]
And I absolutely support this change :)

**Pull request merged**